make_prg/interval_partition.py

```diff
@@ -18,22 +18,24 @@ class IntervalType(Enum):
     Match = auto()
     NonMatch = auto()
 
-
-def get_type(letter: str) -> IntervalType:
-    if letter == "*":
-        return IntervalType.NonMatch
-    else:
-        return IntervalType.Match
+    @classmethod
```

Comment:
For future reference @bricoletc this can just be `@staticmethod`. Random example: https://github.com/mbhall88/MemoryUnits/blob/18551958dbf51a9f68e8733ba17b880dbecc0da7/memory_units.py#L45